tests/api/test_group_api.py

```diff
@@ -0,0 +1,51 @@
+import json
+import unittest
+import requests
+from fauxfactory import gen_string
+from insights.config import Settings
+from insights.group_functions import add_group_api
+from insights.vm_provisioning import VirtualMachine
+
+
+
+class GroupApi(unittest.TestCase):
+    def setup_class(self):
+        self.vm = VirtualMachine()
+        self.vm_name = 'vm_{0}'.format(gen_string('alpha', 6))
+        self.vm.create_openstack_instance(instance_name=self.vm_name, image_name='RHEL-7.1-x86_64',
+                                          flavor_name='Tiny', key_name='jenkins-key', pool_name='public')
+
+        self.setting = Settings()
+        self.session = requests.session()
+        self.session.cert = self.setting.get_certs()
+        self.session.verify = False
+        self.base_url = self.setting.get('api', 'url')
+
+    def test_group_api(self):
+        self.vm.rhsm_register(distro='rhel7')
+        self.vm.register_to_insights()
+        self.display_name = gen_string('alpha', 8)
+
+        # Create group
+        print self.display_name
+        self.payload = add_group_api(self.display_name)
+        self.create_group = self.session.post(self.base_url + '/v1/groups',
+                                              json=self.payload)
+        self.result = self.create_group.text
+        print self.result
+        print self.create_group.status_code
+        print "Created"
+
+        # get group_id
+        response = json.loads(self.result)
+        self.group_id = response['id']
+        print self.group_id
+
+        # Delete group
+        self.delete_group = self.session.delete(self.base_url + '/v1/groups/' + str(self.group_id))
+        print self.delete_group.text
+        print self.delete_group.status_code
+        print "Deleted"
+        self.vm.unregister_from_rhi()
```

Comment:
how is creation and deletion of VM helpful while you are checking the api end points for group creation and deletion.